src/extension.ts

```diff
@@ -0,0 +1,140 @@
+import { from } from 'rxjs'
+import { filter, switchMap } from 'rxjs/operators'
+import * as sourcegraph from 'sourcegraph'
+import { resolveSettings, Settings } from './settings'
+
+/**
+ * Params derived from the document's URI.
+ */
+interface Params {
+    repo: string | null
+    file: string | null
+}
+
+interface SentryProject {
+    projectId: string
+    lineMatches: RegExp[]
+}
+
+const DECORATION_TYPE = sourcegraph.app.createDecorationType()
+const SETTINGSCONFIG = resolveSettings(sourcegraph.configuration.get<Settings>().value)
+
+export function activate(context: sourcegraph.ExtensionContext): void {
+    sourcegraph.workspace.onDidOpenTextDocument.subscribe(textDocument => {
```

Comment:
Why do you need this? You could just use `activeEditor`

Reply:
I'll refactor that part of the code in a separate PR.